### DISCUSSION TITLE
WT32-ETH01 packet loss during ping on the LAN port

### DISCUSSION THREAD

**Discussion by EmilianoMorello** [arendst/Tasmota#21452] [Q&A]
I have a WT32_ETH01 v.1.4 device on which I have loaded the Tasmota 14.0 firmware and configured the template to activate the LAN port. Strangely, I have noticed that pinging the WiFi interface does not lose any packets, while pinging the Ethernet interface loses several packets. Has anyone else had the same problem? Solutions? Thanks.

**Comment by Noschvie** [comment]
Have a look to these commands:

EthType 0, EthClockMode 0 and EthAddress 1

**Reply by EmilianoMorello** [reply]
EthClockMode 0 it worked! Thanks

**Comment by raptorecki** [comment]
Sorry for digging out this old thread, but I'm facing the same problem. 

Tasmota Version 14.3.0 (release-tasmota32)-3_1_0. 
Hardware WT32-ETH01 v1.4. - ESP32-D0WD-V3 v3.1 

Problem with Ethernet connection being unstable on _EthClockMode: 1_ and unavialable on _EthClockMode: 0_.

When I have it with:

```
22:52:26.639 RSL: RESULT = {"Ethernet":"ON"}
22:52:26.872 RSL: RESULT = {"EthAddress":1}
22:52:27.072 RSL: RESULT = {"EthClockMode":1}
22:52:27.272 RSL: RESULT = {"EthType":0}
```

At least it somewhat works. The link and activity LEDs are flashing, it shows:

```
00:00:01.697 ETH: Connected at 100Mbps Full Duplex, Mac F8:B3:B7:2B:A2:0B, Hostname tasmota-2BA208-0520-eth
```

And it manages to get a DHCP lease:

```
23:04:46.947 RSL: RESULT = {"EthIPAddress":"0.0.0.0 (192.168.4.56)"}
23:04:47.177 RSL: RESULT = {"EthSubnetmask":"255.255.255.0"}
23:04:47.377 RSL: RESULT = {"EthGateway":"192.168.4.1"}
```

But it is barely alive:

```
# ping 192.168.4.56
PING 192.168.4.56 (192.168.4.56): 56 data bytes
64 bytes from 192.168.4.56: icmp_seq=5 ttl=64 time=0.837 ms
64 bytes from 192.168.4.56: icmp_seq=7 ttl=64 time=0.721 ms
64 bytes from 192.168.4.56: icmp_seq=22 ttl=64 time=0.825 ms
64 bytes from 192.168.4.56: icmp_seq=27 ttl=64 time=0.830 ms
64 bytes from 192.168.4.56: icmp_seq=30 ttl=64 time=0.837 ms
64 bytes from 192.168.4.56: icmp_seq=34 ttl=64 time=0.731 ms
64 bytes from 192.168.4.56: icmp_seq=38 ttl=64 time=0.767 ms
^C
--- 192.168.4.56 ping statistics ---
39 packets transmitted, 7 packets received, 82.1% packet loss
round-trip min/avg/max/stddev = 0.721/0.792/0.837/0.048 ms
```

But when I switch it to _EthClockMode 0_ the network interface doesn't even light up.

```
22:55:53.981 RSL: RESULT = {"Ethernet":"ON"}
22:55:54.213 RSL: RESULT = {"EthAddress":1}
22:55:54.414 RSL: RESULT = {"EthClockMode":0}
22:55:54.614 RSL: RESULT = {"EthType":0}
```

I tried manual IP settings, wiresharking what it is doing (obviously it does nothing), stock and self compiled binaries and I'm out of ideas. In both cases wifi works just fine.

Have anyone managed to get it to work? I was kind of hoping for those WT32-ETH01 boards to be the cheapest alternative to Olimex ESP32-POE boards (which I had zero problems with) but something isn't quite right here...

Cheers

**Reply by sfromis** [reply]
There were some recent changes potentially affecting how Ethernet works, I'd suggest to test if it works better with the previous release, 14.2.0.

I've got no trouble with WT32-ETH01, but mine is version 1.2 instead of version 1.4.

**Reply by raptorecki** [reply]
Thanks for the reply.

I just tried versions 14.2.0, 13.4.0, 12.5.0 and 11.1.0. Every version displays the same behavior.

I also have three of those boards, all behave the same so it doesn't seems like a unit defect.

**Reply by EmilianoMorello** [reply]
Hi, I have WT32-ETH01 v.1.4 with tasmota firmware 14.0 and no packet loss.

**Comment by raptorecki** [comment]
On current version

```
00:00:00.112 Project tasmota - Tasmota-Wt02 Version 14.3.0(release-tasmota32)-3_1_0(2024-10-15T08:21:27)
```

I enabled console debug and it seems that:

```
00:00:01.058 ETH: Bad EthType or init error
(...)
03:36:24.196 RSL: RESULT = {"Ethernet":"ON"}
03:36:24.421 RSL: RESULT = {"EthAddress":1}
03:36:24.670 RSL: RESULT = {"EthClockMode":0}
03:36:24.920 RSL: RESULT = {"EthType":0}
```

So I thought, well maybe it's just a bused Ethernet controller. But not really. I have three of those boards, all displaying the same behavior. And also it does somewhat work with EthClockMode 1. With a heavy packets lost rate but it does actually recognize the controller and communicate.

```
00:00:01.574 ETH: Attempting connection...
00:00:01.576 ETH: Connected at 100Mbps Full Duplex, Mac F8:B3:B7:2B:A2:0B, Hostname tasmota-2BA208-0520-eth
(...)
03:48:24.701 RSL: RESULT = {"Ethernet":"ON"}
03:48:24.926 RSL: RESULT = {"EthAddress":1}
03:48:25.176 RSL: RESULT = {"EthClockMode":1}
03:48:25.426 RSL: RESULT = {"EthType":0}
```

My v1.4 boards are fairly new (as in "just arrived", I have no idea what is the actual production date). Maybe something has changed in the controller being used?

Just for the record, I checked what it does in EthClockMode 2 and 3. The interface is recognized but it flaps.

```
03:52:13.989 ETH: Connected at 100Mbps Full Duplex, Mac F8:B3:B7:2B:A2:0B, Hostname tasmota-2BA208-0520-eth
03:52:15.048 ETH: IPv6 Local fe80::fab3:b7ff:fe2b:a20b
03:52:15.986 ETH: Disconnected
03:52:19.988 ETH: Connected at 100Mbps Full Duplex, Mac F8:B3:B7:2B:A2:0B, Hostname tasmota-2BA208-0520-eth
03:52:21.048 ETH: IPv6 Local fe80::fab3:b7ff:fe2b:a20b
03:52:21.986 ETH: Disconnected
03:52:27.988 ETH: Connected at 100Mbps Full Duplex, Mac F8:B3:B7:2B:A2:0B, Hostname tasmota-2BA208-0520-eth
03:52:29.048 ETH: IPv6 Local fe80::fab3:b7ff:fe2b:a20b
03:52:29.986 ETH: Disconnected
(...)
03:52:34.238 RSL: RESULT = {"Ethernet":"ON"}
03:52:34.468 RSL: RESULT = {"EthAddress":1}
03:52:34.719 RSL: RESULT = {"EthClockMode":2}
03:52:34.969 RSL: RESULT = {"EthType":0}
```

```
03:50:58.882 ETH: Connected at 100Mbps Full Duplex, Mac F8:B3:B7:2B:A2:0B, Hostname tasmota-2BA208-0520-eth
03:51:00.044 ETH: IPv6 Local fe80::fab3:b7ff:fe2b:a20b
03:51:00.880 ETH: Disconnected
03:51:06.882 ETH: Connected at 100Mbps Full Duplex, Mac F8:B3:B7:2B:A2:0B, Hostname tasmota-2BA208-0520-eth
03:51:08.044 ETH: IPv6 Local fe80::fab3:b7ff:fe2b:a20b
03:51:08.880 ETH: Disconnected
03:51:14.881 ETH: Connected at 100Mbps Full Duplex, Mac F8:B3:B7:2B:A2:0B, Hostname tasmota-2BA208-0520-eth
03:51:16.043 ETH: IPv6 Local fe80::fab3:b7ff:fe2b:a20b
03:51:16.880 ETH: Disconnected
(...)
03:51:28.074 RSL: RESULT = {"Ethernet":"ON"}
03:51:28.297 RSL: RESULT = {"EthAddress":1}
03:51:28.547 RSL: RESULT = {"EthClockMode":3}
03:51:28.797 RSL: RESULT = {"EthType":0}
```
